src/app/src/test/kotlin/org/vaccineimpact/orderlyweb/tests/database_tests/WorkflowRunRepositoryTests.kt

```diff
@@ -162,4 +163,206 @@ class WorkflowRunRepositoryTests : CleanDatabaseTests()
+                "user@email.com",
+                Instant.now(),
+                emptyList(),
+                emptyMap()
+            )
+        )
+
+        val result = sut.getWorkflowRunSummaries()
+        assertThat(result.count()).isEqualTo(3)
+        assertThat(result[0].name).isEqualTo("Report three")
+        assertThat(result[1].name).isEqualTo("Report two")
+        assertThat(result[2].name).isEqualTo("Report one")
+    }
+
+    @Test
+    fun `can get all workflow runs for user`()
+    {
+        insertUser("user@email.com", "user.name")
+        insertUser("user2@email.com", "user2.name")
+
+        val now = Instant.now()
+
+        val sut = OrderlyWebWorkflowRunRepository()
+        sut.addWorkflowRun(
+            WorkflowRun(
+                "Interim report",
+                "adventurous_aardvark",
+                "user@email.com",
+                now,
+                emptyList(),
+                emptyMap()
+            )
+        )
+        sut.addWorkflowRun(
+            WorkflowRun(
+                "Final report",
+                "benevolent_badger",
+                "user2@email.com",
+                now,
+                emptyList(),
+                emptyMap()
+            )
+        )
+
+        val result = sut.getWorkflowRunSummaries("user@email.com")
+        assertThat(result.count()).isEqualTo(1)
+        assertThat(result).isEqualTo(
+            listOf(
+                WorkflowRunSummary(
+                    "Interim report",
+                    "adventurous_aardvark",
+                    "user@email.com",
+                    now
+                )
+            )
+        )
+    }
+
+    @Test
+    fun `can get all workflow runs for user using empty name prefix`()
+    {
+        insertUser("user@email.com", "user.name")
+        insertUser("user2@email.com", "user2.name")
+
+        val now = Instant.now()
+
+        val sut = OrderlyWebWorkflowRunRepository()
+        sut.addWorkflowRun(
+            WorkflowRun(
+                "Interim report",
+                "adventurous_aardvark",
+                "user@email.com",
+                now,
+                emptyList(),
+                emptyMap()
+            )
+        )
+        sut.addWorkflowRun(
+            WorkflowRun(
+                "Final report",
+                "benevolent_badger",
+                "user2@email.com",
+                now,
+                emptyList(),
+                emptyMap()
+            )
+        )
+
+        val result = sut.getWorkflowRunSummaries("user2@email.com", "")
+        assertThat(result.count()).isEqualTo(1)
+        assertThat(result).isEqualTo(
+            listOf(
+                WorkflowRunSummary(
+                    "Final report",
+                    "benevolent_badger",
+                    "user2@email.com",
+                    now
+                )
+            )
+        )
+    }
+
+    @Test
+    fun `can get all workflow runs using non-empty name prefix`()
```

Comment:
So I guess the only scenario not being tested here is for non-empty email *and* non-empty name prefix?

Reply:
Done!